SimDataFormats/CaloAnalysis/interface/SimCluster.h

```diff
@@ -0,0 +1,247 @@
+#ifndef SimDataFormats_SimCluster_h
+#define SimDataFormats_SimCluster_h
+
+#include <vector>
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHit.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+
+//
+// Forward declarations
+//
+class SimTrack;
+class EncodedEventId;
+
+/** @brief Monte Carlo truth information used for tracking validation.
+ *
+ * Object with references to the original SimTrack and parent and daughter TrackingVertices.
+ * Simulation with high (~100) pileup was taking too much memory so the class was slimmed down
+ * and copies of the SimHits were removed.
+ *
+ * @author original author unknown, re-engineering and slimming by Subir Sarkar (subir.sarkar@cern.ch),
+ * some tweaking and documentation by Mark Grimes (mark.grimes@bristol.ac.uk).
+ * @date original date unknown, re-engineering Jan-May 2013
+ */
+class SimCluster
```

Comment:
There are probably 30 lines difference between this and CaloParticle
Why not inherit?